src/api/app/policies/package_policy.rb

```diff
@@ -58,4 +58,20 @@ def source_access?
 
     record.enabled_for?('sourceaccess', nil, nil)
   end
+
+  def assign?
+    return false unless Flipper.enabled?(:foster_collaboration, user)
+    return true if user.admin?
+
+    assigneer_is_a_collaborator?
+  end
+
+  private
+
+  def assigneer_is_a_collaborator?
+    collaborators = (record.relationships + record.project.relationships).map(&:user)
+    return false if collaborators.empty?
+
+    collaborators.include?(user)
+  end
```

Comment:
I think this can be moved into User, that way you can reuse it for this and for the AssignmentPolicy